### PR TITLE
Edit permissions crashes & improvements

### PR DESCRIPTION
**Pull request by jelly**
Fixes the 4 issues in https://github.com/cockpit-project/cockpit-files/issues/395 except using `getent passwd` and `getent group` I'll look into that on Monday in https://github.com/cockpit-project/cockpit-files/issues/431

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -47,23 +48,29 @@ export interface FolderFileInfo extends FileInfo {
...
+    addAlert: (title: string, variant: AlertVariant, key: string) => void,
+    cwdInfo: FileInfo | null,
+}
+
 export const FilesContext = React.createContext({
-    // eslint-disable-next-line no-unused-vars
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    addAlert: (title: string, variant: AlertVariant, key: string) => console.warn("FilesContext not initialized")
-});
+    addAlert: () => console.warn("FilesContext not initialized"),
```
Comment (cockpituous) [inline src/app.tsx:57]: This added line is not executed by any test.

File: src/app.tsx [hunk 2]
```diff
@@ -114,55 +125,61 @@ export const Application = () => {
...
+                    <AlertGroup isToast isLiveRegion>
+                        {alerts.map(alert => (
+                            <Alert
+                              variant={alert.variant}
+                              title={alert.title}
+                              actionClose={
+                                  <AlertActionCloseButton
+                                    title={alert.title}
+                                    variantLabel={`${alert.variant} alert`}
+                                    onClose={() => removeAlert(alert.key)}
```
Comment (cockpituous) [inline src/app.tsx:139]: These 9 added lines are not executed by any test.

File: src/app.tsx [hunk 3]
```diff
@@ -114,55 +125,61 @@ export const Application = () => {
...
+                                  />
+                              }
+                              key={alert.key}
```
Comment (cockpituous) [inline src/app.tsx:142]: This added line is not executed by any test.

File: src/app.tsx [hunk 4]
```diff
@@ -48,11 +48,17 @@ export interface FolderFileInfo extends FileInfo {
...
+    addAlert: (title: string, variant: AlertVariant, key: string) => void,
+    cwdInfo: FileInfo | null,
+}
+
 export const FilesContext = React.createContext({
-    // eslint-disable-next-line no-unused-vars
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    addAlert: (title: string, variant: AlertVariant, key: string) => console.warn("FilesContext not initialized")
-});
+    addAlert: () => console.warn("FilesContext not initialized"),
```
Comment (martinpitt) [inline src/app.tsx:57]: I have a very strong déjà vu on this one. Dropping the signature here feels wrong at least for documentation purposes, but also for type checking -- this is very much not a `FilesContextType` now, but really should be. Perhaps prefix the parameter names with `_`, does that work? If not, I'd prefer keeping the eslint overrides.

File: src/fileActions.jsx [hunk 5]
```diff
@@ -268,45 +268,13 @@ const EditPermissionsModal = ({ selected, path }) => {
     const [errorMessage, setErrorMessage] = useState(undefined);
     const accounts = useFile("/etc/passwd", { syntax: etcPasswdSyntax });
     const groups = useFile("/etc/group", { syntax: etcGroupSyntax });
```
Comment (martinpitt) [inline src/fileActions.jsx:270]: As discussed, this needs to be `getent passwd` and `getent group`, but that can be a follow-up.

File: src/fileActions.jsx [hunk 6]
```diff
@@ -310,6 +310,10 @@ const EditPermissionsModal = ({ selected, path }) => {
         ];
     }
 
+    function sortByName(a, b) {
+        return a.name.toLowerCase() < b.name.toLowerCase() ? -1 : 1;
```
Comment (martinpitt) [inline src/fileActions.jsx:314]: Let's use `a.name.localeCompare(b.name)`?

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks, nice cleanups/fixes! 

However, this breaks TestFiles.testContextMenu everywhere, so still needs work. I also have some remarks.

File: src/app.tsx (hunk 4 above)
Comment (jelly) [reply src/app.tsx:57]: Yeah it's about the unused warnings which are super annoying.

File: src/fileActions.jsx (hunk 6 above)
Comment (jelly) [reply src/fileActions.jsx:314]: Nice one, in that case, we should also update the sorting in the files card body

File: src/app.tsx (hunk 4 above)
Comment (jelly) [reply src/app.tsx:57]: But honestly, this is valid JS, it is a valid `addAlert` function as one can check `arguments` and calling addAlert with extra arguments is fine in JavaScript.

I can add it, and add `    // eslint-disable-next-line @typescript-eslint/no-unused-vars` but it feels wrong. The Object is typed as FilesContextType and this still works

image

File: src/fileActions.jsx (hunk 5 above)
Comment (jelly) [reply src/fileActions.jsx:270]: I'll do it as follow up

**Comment by jelly** [comment]
image

This is indeed broken

File: src/app.tsx (hunk 4 above)
Comment (martinpitt) [reply src/app.tsx:57]: Well, I'm not very familiar with type checking, so if you say that's kosher, then so be it. :grin:

File: src/app.tsx (hunk 1 above)
Comment (cockpituous) [inline src/app.tsx:57]: This added line is not executed by any test.

File: src/app.tsx (hunk 2 above)
Comment (cockpituous) [inline src/app.tsx:139]: These 9 added lines are not executed by any test.

File: src/app.tsx (hunk 3 above)
Comment (cockpituous) [inline src/app.tsx:142]: This added line is not executed by any test.

**Review by martinpitt — APPROVED** [review]
Thanks!

**Pull request merged**